src/main/java/com/gradle/enterprise/api/ApiProblemParser.java

```diff
@@ -9,18 +9,18 @@
 import java.util.Optional;
 
 public final class ApiProblemParser {
-
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
     private static final String CONTENT_TYPE = "application/problem+json";
 
-    public static Optional<ApiProblem> maybeParse(ApiException apiException) {
+    public static Optional<ApiProblem> maybeParse(ApiException apiException, ObjectMapper objectMapper) {
         return apiException.getResponseHeaders()
-            .firstValue("content-type")
-            .filter(contentType -> contentType.startsWith(CONTENT_TYPE))
+            .get("content-type")
+            .stream()
+            .findFirst()
+            .filter(headerValue -> headerValue.startsWith(CONTENT_TYPE))
             .map(__ -> {
                 try {
-                    return OBJECT_MAPPER.readValue(apiException.getResponseBody(), ApiProblem.class);
-                } catch (JsonProcessingException e) {
+                    return objectMapper.readValue(apiException.getResponseBody(), ApiProblem.class);
+                } catch (final JsonProcessingException e) {
```

Comment:
why final ?

Reply:
I just reverted the code back to what it was, but I agree that final is overkill here.

Reply:
@welandaz any reason for adding the `final`? https://github.com/gradle/gradle-enterprise-api-samples/commit/5ac38f43d780240d539a1130613101711c7eecb2#diff-c09db6258cb73d6d349a5706124a5b0308704c707c0e04c1e5c8c9c1b1a8cd22R24

Reply:
Since we don't have a convention to use `final` when variable is not being changed, feel free to remove it.